include/Platform/Universal Windows/UWP/RTHelpers.h

```diff
@@ -893,22 +893,19 @@ static HString nsStrToHstr(NSString* str) {
         // We increment this just to make sure that we return an empty HString.
         dataBytes++;
 
-        // Include size of the additional terminating '\0' character.
-        data.reserve(dataBytes + 1);
+        // Allocate a vector containing only the null character.
+        data.resize(dataBytes, L'\0');
 
-        // Since the string is empty, assign the terminating character
-        // right at the start.
-        data[0] = L'\0';
     } else {
         // Include size of the terminating '\0' character.
         data.reserve(dataBytes + 1);
 
-        [str getCharacters : &data[0]];
+        [str getCharacters : data.data()];
+        
+        // Add terminating NULL character.
+        data[dataBytes] = L'\0';
     }
 
-    // Add terminating NULL character.
-    data[dataBytes] = L'\0';
-
     HRESULT hr = WindowsCreateString((LPWSTR)&data[0], dataBytes, ret.GetAddressOf());
```

Comment:
It looks like, when the source string is empty, we're creating a length-1 HSTRING containing a single embedded NUL character. That doesn't seem like the right behavior, since that's not the same as an empty string. In particular, WindowsIsStringEmpty will return FALSE for such a string, and WindowsGetStringLen will return 1.

I see the comment that says we're trying to avoid having WindowsCreateString return a null pointer for empty strings. But why are we trying to avoid that?